components/nexus-core/src/main/java/org/sonatype/nexus/internal/httpclient/DefaultsCustomizer.java

```diff
@@ -49,7 +51,8 @@
   @Inject
   public DefaultsCustomizer(
       final UserAgentGenerator userAgentGenerator,
-      @Named("${nexus.httpclient.requestTimeout:-30s}") final Time requestTimeout,
+      @Named("${nexus.httpclient.requestTimeout:-20s}") final Time requestTimeout,
+      @Named("${nexus.httpclient.connectionRequestTimeout:-30s}") final Time connectionRequestTimeout,
```

Comment:
After looking at httpclient source code, I understand this to be the time to wait for getting a connection from the connection manager impl.

Is 30s the same as Nexus 2?

I wish the name of the property could make this clearer. 'connectionRequestTimeout' sounds like a timeout when the connection makes a request, instead of the 'timeout for getting a connection from the connection manager'.

maybe something like 'connectionAvailabilityTimeout' ?

The lack of javadoc here seems not be helping the intent of the param.